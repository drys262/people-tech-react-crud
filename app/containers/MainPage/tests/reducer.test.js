// import produce from 'immer';
import mainPageReducer from '../reducer';
import {
  toggleFilter,
  loadPeople,
  loadPeopleFailed,
  loadPeopleSuccess,
} from '../actions';

/* eslint-disable default-case, no-param-reassign */
describe('mainPageReducer', () => {
  it('returns the initial state', () => {
    expect(mainPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the toggleFilter', () => {
    expect(mainPageReducer({}, toggleFilter())).toMatchSnapshot();
  });

  it('handles the loadPeople', () => {
    expect(mainPageReducer({}, loadPeople(1))).toMatchSnapshot();
  });
  it('handles the loadPeopleFailed', () => {
    expect(
      mainPageReducer({}, loadPeopleFailed('error message')),
    ).toMatchSnapshot();
  });
  it('handles the loadPeopleSuccess', () => {
    expect(mainPageReducer({}, loadPeopleSuccess([]))).toMatchSnapshot();
  });
});
