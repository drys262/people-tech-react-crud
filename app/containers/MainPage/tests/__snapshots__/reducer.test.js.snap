// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainPageReducer handles the loadPeople 1`] = `
Object {
  "isFetchingPeople": true,
}
`;

exports[`mainPageReducer handles the loadPeopleFailed 1`] = `
Object {
  "isFetchingPeople": false,
}
`;

exports[`mainPageReducer handles the loadPeopleSuccess 1`] = `
Object {
  "isFetchingPeople": false,
  "people": Array [],
}
`;

exports[`mainPageReducer handles the toggleFilter 1`] = `
Object {
  "toggleFilter": "asc",
}
`;

exports[`mainPageReducer returns the initial state 1`] = `
Object {
  "error": false,
  "errorMessage": "",
  "isFetchingPeople": false,
  "loadingRepos": false,
  "people": Array [],
  "searchTerm": "",
  "selectedDev": Object {},
  "selectedDevRepo": Array [],
  "toggleFilter": "asc",
}
`;
